### PR TITLE
feat(codex): add `codemem setup --codex` for plugin-free install

### PR DESCRIPTION
**Pull request by kunickiaj**
## Description
Adds `codemem setup --codex` — a plugin-free way to configure Codex MCP + hooks by writing Codex's own config files. This unblocks **API-key / non-subscription Codex Desktop**, where the plugin marketplace (the normal install path) is greyed out.

What it does (idempotent; honors `CODEX_HOME`; backs up before changing; `--force` refreshes codemem-owned entries):
- **MCP:** appends `[mcp_servers.codemem]` (`command = "npx"`, `args = ["-y", "codemem", "mcp"]`) to `<CODEX_HOME>/config.toml` only when absent. The file is **never reparsed or reformatted** — only appended — so comments and unrelated servers (including secrets) are preserved. Detection uses a whitespace/quote-tolerant regex that won't false-match siblings like `[mcp_servers.codemem-foo]`.
- **Hooks:** merges `SessionStart`, `UserPromptSubmit` (ingest **and** inject), `PostToolUse`, `Stop` into `<CODEX_HOME>/hooks.json` (Codex's User-layer hooks file), preserving any unrelated user hooks.
- **Command resolution:** hooks call `codemem codex-hook-*` directly when `codemem` is on `PATH` (fast; timeouts 10s), else fall back to `npx -y codemem codex-hook-*` (timeouts 30/20 to absorb a cold npx). Mirrors the plugin wrapper's direct-first / npx-fallback model.
- Wiring: new `--codex-only`; a plain `codemem setup` also configures Codex when `~/.codex` (or `$CODEX_HOME`) is detected. OpenCode/Claude paths unchanged.

Hooks from the user config layer require a one-time trust approval in Codex (prompted on first run); MCP recall needs no trust.

## Type of Change

- [x] 🚀 Feature (new functionality)
- [ ] 🐛 Bug fix (fixes an issue)
- [ ] 📚 Documentation (docs-only change)
- [ ] 🔧 Maintenance (refactor, chore, CI, etc.)
- [ ] 🧪 Testing (test-only changes)

## Testing

- [x] `pnpm run tsc`, `pnpm run lint` (469 files), `pnpm run test` (2265 pass)
- [x] 16 unit tests in `setup-codex.test.ts`: fresh install schema, idempotency, non-destructive merge (config.toml comments/other servers + unrelated hooks), `--force` preserve-unrelated, backups, regex edge cases (`codemem-foo` not matched, quoted/whitespace matched), malformed-json no-clobber, direct vs npx command base
- [x] Real smoke against a temp `CODEX_HOME`: preserves existing config + appends MCP; writes hooks.json (UserPromptSubmit ingest+inject in order); resolves **direct** `codemem` commands when on PATH; idempotent re-run; backup created

## Checklist

- [x] Code follows project style (`pnpm run lint` passes for touched files)
- [x] Self-review completed + CodeReviewer pass (ship-blocker on command resolution addressed: direct-when-on-PATH + npx fallback)
- [x] Documentation updated (README + docs/plugin-reference.md plugin-free path)
- [x] No new warnings introduced

### CODE REVIEW

**Comment by kunickiaj** [comment]
* **#1227** <a href="https://app.graphite.com/github/pr/kunickiaj/codemem/1227?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/kunickiaj/codemem/1227?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/cli/src/commands/setup.ts [hunk 1]
```diff
@@ -289,33 +295,322 @@ function installClaudeMcp(force: boolean): boolean {
...
+			(h as { command: string }).command.includes(CODEMEM_HOOK_MARKER),
+	);
+}
+
+/** Detect whether `codemem` resolves on PATH. */
+function codememOnPath(): boolean {
+	try {
+		execFileSync(process.platform === "win32" ? "where" : "which", ["codemem"], {
+			stdio: "ignore",
+		});
```
[the remark this answers is not in the document]
Comment (kunickiaj) [reply packages/cli/src/commands/setup.ts]: Fixed (P1). `codememOnPath()` now resolves the actual `which`/`where` path and treats a transient npx/dlx cache bin (`.../_npx/...` or `.../.pnpm/dlx/...`) as NOT on PATH, so `npx -y codemem setup --codex` correctly bakes `npx -y codemem codex-hook-*` into hooks.json (which still resolves after npx exits). A durable global install still gets direct `codemem` commands. Added `isTransientNpxBinPath` with unit tests and a smoke under a simulated npx PATH (commit abb06456).

File: packages/cli/src/commands/setup.ts [hunk 2]
```diff
@@ -289,33 +295,322 @@ function installClaudeMcp(force: boolean): boolean {
...
-		}
-
-		if (ok) {
-			p.outro("Setup complete — restart your editor to load the plugin");
-		} else {
-			p.outro("Setup completed with warnings");
-			process.exitCode = 1;
-		}
-	});
+	.option("--codex-only", "only install for Codex")
```
[the remark this answers is not in the document]
Comment (kunickiaj) [reply packages/cli/src/commands/setup.ts:586]: Fixed (P2). Added `--codex` as a documented alias for `--codex-only` (the action treats them identically). Added a test asserting both options are declared (commit abb06456).
